dwave/plugins/sklearn/transformers.py

```diff
@@ -198,27 +177,127 @@ def correlation_cqm(
-            # entries for each correlation coefficient) is retained for consistency with
-            # the original formulation from Milne et al.
+            # we multiply by num_features to have consistent performance 
+            # with the increase of the number of features
+            np.fill_diagonal(correlations, correlations[:, -1] * (- alpha * num_features))
+            # Note: we only add terms on and above the diagonal
             it = np.nditer(correlations[:-1, :-1], flags=['multi_index'], op_flags=[['readonly']])
+            print(correlations)
             cqm.set_objective((*it.multi_index, x) for x in it if x)
 
         return cqm
 
+    @staticmethod
+    def mutual_information_cqm(        
+        X: npt.ArrayLike,
+        y: npt.ArrayLike,
+        *,
+        num_features: int,
+        alpha: float = 0.5,
+        strict: bool = True,
+        conditional: bool = True,
+        discrete_features: typing.Union[str, npt.ArrayLike] = "auto",
+        discrete_target: bool = False,
+        copy: bool = True,
+        n_neighbors: int = 4,
+        n_workers: int = 1,
+        random_state: typing.Union[None, int] = None,
+        ) -> dimod.ConstrainedQuadraticModel:
+        """Build a constrained quadratic model for feature selection.
+
+        If ``conditional`` is True then the conditional mutual information 
+        criterion from [2] is used, and if ``conditional`` is False then
+        mutual information based criterion from [1] is used.
+        
+        For computation of mutual information and conditional mutual information 
+        
+        
+        Args:
+            X:
+                Feature vectors formatted as a numerical 2D array-like.
+            y:
+                Class labels formatted as a numerical 1D array-like.
+            alpha:
+                Hyperparameter between 0 and 1 that controls the relative weight of
+                the relevance and redundancy terms.
+                ``alpha=0`` places no weight on the quality of the features,
+                therefore the features will be selected as to minimize the
+                redundancy without any consideration to quality.
+                ``alpha=1`` places the maximum weight on the quality of the features,
+                and therefore will be equivalent to using
+                :class:`sklearn.feature_selection.SelectKBest`.
+            num_features:
+                The number of features to select.
+            strict:
+                If ``False`` the constraint on the number of selected features
+                is ``<=`` rather than ``==``.
+            conditional: bool, default=True
+                Whether to compute the off-diagonal terms using the conditional mutual
+                information or joint mutual information
+            discrete_features: 
+                See :func:`sklearn.feature_selection._mutual_info._estimate_mi`
```

Comment:
I am a bit nervous using an internal function in an internal module....

Three options:
1) Ask `scikit-learn` to expose these methods publicly. I don't see a issue open to do so right now, we could/should open one
2) Port the code directly into our library, so there is no danger of them changing it our from under us
3) Continue using a private method/module.

I think we should do (1) regardless.

Also, we don't build the docs for this package so it doesn't really matter, but this link won't work because there are no docs published by sklearn for this method.

Reply:
found a quick solution, but in some cases extra computations is done, which potentially could be avoided.